### DISCUSSION TITLE
404 NOT_FOUND ERROR

### DISCUSSION THREAD

**Discussion by thinkux-dev** [vercel/community#5715] [Help] [answered]
### Summary

Hello, been trying to deploy a project but keep getting the  "404: NOT_FOUND Code: NOT_FOUND ID: cdg1::xdwlw-1706844949255-2c99a53fa9f7"  Being having problems with this tried finding solutions but all seem to not work.  Here is my vercel.json file configuration. "{   "version": 2,   "routes": [   { "src": "/(.)", "dest": "/server.js" },     { "src": "/views/layouts/main.handlebars", "dest": "/server.js" },     { "src": "/public/(.*)", "dest": "/public/$1" }   ] }"  Don't know if I am doing something wrong. Please can you help out?

### Example

https://puzzle-africa-com.vercel.app/

### Steps to Reproduce

"{   "version": 2,   "routes": [   { "src": "/(.)", "dest": "/server.js" },     { "src": "/views/layouts/main.handlebars", "dest": "/server.js" },     { "src": "/public/(.*)", "dest": "/public/$1" }   ] }"

**Comment by ATK422** [comment]
I am also getting the same issue on cle1, glad to see its not just me.

Adding on to this a bit, preview domains work fine, but the actual main vercel domain doesn't work.

**Comment by ATK422** [comment]
Seems to be resolved now

**Reply by thinkux-dev** [reply]
What did you do please?

**Reply by ATK422** [reply]
I just waited and it fixed itself, no clue what was wrong.

**Reply by dheerajaggarwal2** [reply]
I am also getting the same error. Is there any solution for it? I am deploying an express code
https://testing-bansal-backend-service.vercel.app/

**Reply by Khuramjaffery** [reply]
404: NOT_FOUND
Code: NOT_FOUND
ID: bom1::qx2bl-1711731863634-7215ba23710f

**Reply by gautamalish** [reply]
did you fix it? I'm getting the same error

**Comment by amyegan** [comment]
Do you have a public repo or [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) to show how your project is designed? The correct configuration depends on the app structure.

**Reply by thinkux-dev** [reply]
I think my wrong was not using react or next for it. I got some advice from a friend and he said that Vercel only supports react/next or nuxt. So I rewrote the application using React and was able to deploy it on Vercel successfully.

**Reply by amyegan** [reply]
I'm glad you found a solution 😄

**Reply by kartikayakul01** [reply]
@amyegan did you get any solution? I'm trying to solve it but don't what is the actual problem
"dependencies": {
    "@reduxjs/toolkit": "^2.2.3",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-icons": "^5.0.1",
    "react-redux": "^9.1.0",
    "react-router-dom": "^6.22.3"
  }
only front end used

**Reply by zaindunnenterprisesinc** [reply]
@amyegan  hey i am getting the error can you please help me ?

**Comment by Khuramjaffery** [comment]
404: NOT_FOUND
Code: NOT_FOUND
ID: bom1::qx2bl-1711731863634-7215ba23710f

 guys   help and resolve the error

**Reply by kartikayakul01** [reply]
Is it resolved ? I'm also getting same error please help you have solved.

**Comment by mwakazi-dev** [comment]
Hello guys?

Has anyone found a solution?

**Reply by mwakazi-dev** [reply]
Getting the same error:

404: NOT_FOUND
Code: NOT_FOUND
ID: bom1::62p2f-1715780769671-36f9cb74f4c7

**Reply by sbti-de** [reply]
I am still getting 404: NOT_FOUND
Code: DEPLOYMENT_NOT_FOUND error for my Reactjs developed app, are you still facing same error, since yesterday ?

**Reply by thinkux-dev** [reply]
@sbti-de  and @jmwakz99 have you been able to resolve it? if no, can i see your vercel.js file?

**Comment by MohammedMustafaInf** [comment]
sss

**Comment by Suriyakuamr** [comment]
404: NOT_FOUND
Code: NOT_FOUND
ID: bom1::jr52j-1716365856692-25dcaf063c8f 

I AM FACING ERROR PLEASE SOLVE IT.

**Reply by thinkux-dev** [reply]
have you been able to resolve it? if not, can I see your vercel.js file?

**Reply by Suriyakuamr** [reply]
This is my file vercel.json. the main file is cart.py.                                                                 {
    "version": 2,
    "builds": [
        {
            "src": "cart.py",
            "use": "@vercel/python"
        }
    ],
    "routes": [
        {
            "src": "/(.*)",
            "dest": "cart.py"
        }
    ]
}

**Comment by Kishokkumar2** [comment]
404: NOT_FOUND
Code: NOT_FOUND
ID: bom1::w2rk9-1716494374327-6e743a3575fa

Any one can help me to solve this fucking error

**Reply by mwakazi-dev** [reply]
Is this a next js project?

**Reply by mwakazi-dev** [reply]
If so add vercel.json with the following contents.

```
{
  "framework": "nextjs"
}

```

**Comment by amyegan** [comment, marked as answer]
A [404 NOT_FOUND](https://vercel.com/docs/errors/NOT_FOUND#not_found) means the requested resource could not be found. While there are too many possible causes for me to be able to offer and exact solution for everyone, here are some things you can check to narrow down the cause in your case.

- Check the deployment URL for typos or incorrect paths.
- Make sure the [deployment exists](https://vercel.com/docs/projects/project-dashboard#deployments) and has not been deleted.
- If the deployment exists, review the [deployment logs](https://vercel.com/docs/deployments/logs) for issues that might have caused the deployment to be unavailable.
- From the deployment details page > **Source** tab, toggle to **Output**, and verify that an **index** is available. This is sometimes missing. I've seen it happen with some Express.js setups, for example, and I would recommend the [Express on Vercel](https://vercel.com/guides/using-express-with-vercel) guide in those cases.
- Check the [Runtime Logs](https://vercel.com/docs/observability/runtime-logs#runtime-logs) for errors preventing pages from being available.
- If you're on a team with multiple people, make sure you have [permissions](https://vercel.com/docs/accounts/team-members-and-roles) to access the deployment.

https://vercel.com/docs/errors/NOT_FOUND#not_found

**Reply by prakash855** [reply]
shittiest answer ever.
I passed all the checks and still I am getting the error

**Reply by amyegan** [reply]
Sorry you feel that way! Without knowing the details of your project I can't offer a more specific solution. There are a lot of reasons why a file might be at a different path than what you expected or might not exist in the deployment at all. 

Do you have a [minimal reproducible example](https://vercel.com/guides/creating-a-minimal-reproducible-example) that we can use to help you debug?

**Comment by ldhoax** [comment]
Here is my Vercel project with 3 domains
![image](https://github.com/vercel/community/assets/41488436/17057f15-9450-44bd-98c6-2970d4d247b8)
In Cloudflare, I added a CNAME record, with name is my domain name (example.com) and content is one of three domains in Vercel above, also set it as DNS only
![image](https://github.com/vercel/community/assets/41488436/b6cbd0b3-534a-4ecc-8426-4e8092033920)
But, I got DEPLOYMENT_NOT_FOUND when access my url (example.com)
![image](https://github.com/vercel/community/assets/41488436/a3101cf6-beeb-4966-9270-a15d6c798dd2)
I had tried with all three Vercel domains for Cloudflare DNS record. I also read Vercel document and re-checked my url, domains too but I can confirm they're correct.
Am I going wrong somewhere :thinking: ?

**Reply by amyegan** [reply]
Try adding the domain to your project first.

https://vercel.com/docs/getting-started-with-vercel/domains#add-a-domain

**Reply by ldhoax** [reply]
It's work, thanks a lot @amyegan :pray:

**Comment by iam-abhishek-yadav** [comment]
Add a file **vercel.json** in your root directory. This worked for me.
```json
{
  "rewrites":  [
    {"source": "/(.*)", "destination": "/"}
  ]
}
```

**Reply by pawlean** [reply]
Thanks for sharing this!

**Comment by HouWin** [comment]
404: NOT_FOUND
Code: NOT_FOUND
ID: hnd1::rjnv4-1721187980425-15ff538a9576
Local preview is normal, I don't know how to handle this issue

**Reply by iam-abhishek-yadav** [reply]
Try adding a vercel.json to root folder
```json
{
  "rewrites":  [
    {"source": "/(.*)", "destination": "/"}
  ]
}
```

**Reply by diogorh** [reply]
When I add this rewrite, my endpoint returns the index.html page. 
In my localhost it works just fine. This error is driving me crazy.

**Reply by FullStack02002** [reply]
thanks man
it really worked

**Reply by pawlean** [reply]
We also wrote a post in the new forum which may be helpful: https://vercel.community/t/debugging-404-errors/437

[a comment by an automated account is not shown]
